src/app/ui/team-page/team-page.component.ts

```diff
@@ -1,20 +1,38 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { AuthService } from 'src/app/core/auth.service';
-import { Observable, combineLatest } from 'rxjs';
+import { Observable } from 'rxjs';
 import { SeasonService } from 'src/app/core/season.service';
+import { TeamsService } from '../../teams/teams.service';
+import { take, tap, flatMap } from 'rxjs/operators';
 
 @Component({
   selector: 'team-page',
   templateUrl: './team-page.component.html',
   styleUrls: ['./team-page.component.scss']
 })
-export class TeamPageComponent {
+export class TeamPageComponent implements OnInit {
   seasonId$: Observable<string>;
+  teams$: Observable<Membership[]>;
+  selectedTeam: Membership;
+  compareFn: ((f1: Membership, f2: Membership) => boolean) | null = this.compareByValue;
 
-  constructor(
-    public auth: AuthService,
-    public seasonService: SeasonService
-    ) {
+  constructor(public auth: AuthService, public seasonService: SeasonService, public teamService: TeamsService) {
     this.seasonId$ = this.seasonService.getEnabledSeasonId();
   }
+
+  ngOnInit() {
+    this.selectedTeam = null;
+    // getting the list of teams the user is a member of
+    this.teams$ = this.auth.user$.pipe(
+      take(1),
+      tap((user: User) => {
+        this.selectedTeam = user.membership;
+      }),
+      flatMap((user) => this.teamService.getAllMemberships(user.uid))
+    );
+  }
+
+  compareByValue(f1: Membership, f2: Membership) {
```

Comment:
maybe also change its name to be a bit more relevant e.g.:
```suggestion
  compareById(f1: Membership, f2: Membership): boolean {
```

Reply:
this has also been covered.